### PR TITLE
feat: add dependency version checks for package.json

### PR DESCRIPTION
**Pull request by janbiasi**
### Why is this feature required?
If we check that `save-exact` is enabled, we only ensure that no-one is able to install an approximate version via the CLI, but it's still possible to have direct invalid dependency version definitions inside the `package.json` which is still one of the pain points in regards to the whole save-exact topic. Additionally if we integrate the linter in an older project it would tell us right ahead if there are still some invalid version declarations inside the `package.json`.

### What is the content of the change?
A new flag `-d` / `--dependenciesExactVersion` flag can be use, which will check the `package.json` in the repository (or all `package.json` in a monorepository) against invalid definitions (tarball, approximate version or the star wildcard). The readme has also been updated.

### Other changes
Adjusted the process exit code on uncaught issues to `2` as we don't want the same exit code for an expected error and unexpected error.

**Breaking change**: no
**API changes**: new flag `-d`, `--dependenciesExactVersion`
**Monorepo compatible**: yes

### CODE REVIEW

**Comment by smollweide** [comment]
- new flag dependencies-exact-version
- check also monorepo packages

**Comment by janbiasi** [comment]
@smollweide & @jegli changes done, could you please check the PR again? 😃

File: package.json [hunk 1]
```diff
@@ -71,6 +72,9 @@
     "webpack-cli": "3.3.10",
     "zen-observable": "0.8.15"
   },
+  "dependencies": {
+    "find-packages": "7.0.2"
```
Comment (smollweide) [inline package.json]: that's a devDependency because everything will be bundled into on file by using webpack.

Comment (janbiasi) [reply package.json]: ah okay didn't know that as it's not that common for server modules, I'll move that dependency asap., thanks!

**Comment by jegli** [comment]
Thank you @janbiasi for this amazing contribution! I fixed a typo and updated the `find-packages` dependency from 7.0.2 to 7.0.6. Unfortunately, the project fails to build / test now which confuses me slightly. 😅

I have two other questions:
- [process.exit(2)](https://github.com/namics/env-linter/pull/6/files#diff-45ec1a2ac33cf965ab5afde0254bd928L48) Why did you change this?
- Would you generally recommend using JSDocs for a project like this? So far there are no JSDocs but I guess I could update the existing code.

Oh and sorry that I took me so long to look at this PR.

**Comment by jegli** [comment]
`7.0.4` seems to be the latest that works. Between that version and `7.0.5` the definition-file has changed as follows:
```
declare function findPkgs(root: string, opts?: findPkgs.Options): Promise<findPkgs.Project[]>;
export = findPkgs;
```
to:
```
export default function findPkgs(root: string, opts?: findPkgs.Options): Promise<findPkgs.Project[]>;
export {};
```

Is there a way I can still get the [Project interface](https://github.com/namics/env-linter/pull/6/files#diff-249a34e7290e0ff8fc38bfb3c55359ffR90)?

**Comment by jantimon** [comment]
You should probably ask them to fix that for their project - (maybe even provide a pull request and explain why you need the typed return value)

If they won't be willing to help you can use `ReturnType<findPkgs>`

**Comment by jegli** [comment]
I used the [typings from pnpm](https://github.com/pnpm/pnpm/tree/master/packages/types#readme) (PackageManifest) and added the [IProject interface](https://github.com/namics/env-linter/pull/6/commits/7463682abbbbf6b0856565c2dbd27917ac8e7743). I hope that makes sense.

I [excluded node_modules from findPackages](https://github.com/namics/env-linter/pull/6/commits/c7d3c59d24aebf7c3029f48d8cb4ceb9f73832e5), since we don't want to look through that whole tree and I made the [check usable by options](https://github.com/namics/env-linter/pull/6/commits/468a7064d3a1b1110626f01cffe11d221afb383f).

I want to test a little more and make it a little less verbose for mono-repos but hopefully we can merge this soon.
Screenshot 2020-05-25 at 13 44 41

**Comment by jegli** [comment]
I added the name of the package to the success-message, so that it gives a little more context. 
Screenshot 2020-05-25 at 17 21 48

**Pull request merged**